tests/components/RouterTest.php

```diff
@@ -67,29 +69,99 @@ public function testRunThrowsLogicExceptionOnEmptyInput():void {
+                            "chemicals/list"                  => "index",
+                            "chemicals/add"                   => "add",
+                            "chemicals/edit/([1-9][0-9]*$)"   => "edit",
+                            "chemicals/delete/([1-9][0-9]*$)" => "delete"
+                        ),
+                        "CompaniesFactory" => array(
+                            "companies/list"                  => "index",
+                            "companies/add"                   => "add",
+                            "companies/edit/([1-9][0-9]*$)"   => "edit",
+                            "companies/delete/([1-9][0-9]*$)" => "delete"
+                        ),
+                        "CountriesFactory" => array(
+                            "countries/list"                  => "index",
+                            "countries/add"                   => "add",
+                            "countries/edit/([1-9][0-9]*$)"   => "edit",
+                            "countries/delete/([1-9][0-9]*$)" => "delete"
+                        ),
+                        "GendersFactory" => array(
+                            "genders/list"                  => "index",
+                            "genders/add"                   => "add",
+                            "genders/edit/([1-9][0-9]*$)"   => "edit",
+                            "genders/delete/([1-9][0-9]*$)" => "delete"
+                        ),
+                        "MedicinesFactory" => array(
+                            "medicines/list"                  => "index",
+                            "medicines/add"                   => "add",
+                            "medicines/edit/([1-9][0-9]*$)"   => "edit",
+                            "medicines/delete/([1-9][0-9]*$)" => "delete"
+                        ),
+                        "PurposesFactory" => array(
+                            "purposes/list"                  => "index",
+                            "purposes/add"                   => "add",
+                            "purposes/edit/([1-9][0-9]*$)"   => "edit",
+                            "purposes/delete/([1-9][0-9]*$)" => "delete"
+                        ),
+                        "RestrictsFactory" => array(
+                            "restricts/list"                  => "index",
+                            "restricts/add"                   => "add",
+                            "restricts/edit/([1-9][0-9]*$)"   => "edit",
+                            "restricts/delete/([1-9][0-9]*$)" => "delete"
+                        ),
+                        "UsersFactory" => array(
+                            "users/list"      => "index",
+                            "users/register"  => "register",
+                            "users/authorise" => "authorise",
+                            "users/logout"    => "logout"
+                        )
+                    )
+                ]
+            ];
         }
```

Comment:
Array of arrays. The indexes of the outer array are names of Factories, indexes of the inner arrays are routes and the values are Controller actions.